tests/transformers/test_transformers_model_export.py

```diff
@@ -82,6 +85,36 @@
 # - Text2TextGeneration pipeline tests
 # - Conversational pipeline tests
 
+_logger = logging.getLogger(__name__)
+
+
+def flaky(max_tries=3):
+    """
+    Annotation decorator for retrying flaky functions up to max_tries times, and raise the Exception
+    if it fails after max_tries attempts.
+    :param max_tries: Maximum number of times to retry the function.
+    :return: Decorated function.
+    """
+
+    def flaky_test_func(test_func):
+        @wraps(test_func)
+        def decorated_func(*args, **kwargs):
+            error = None
+            for i in range(max_tries):
+                try:
+                    return test_func(*args, **kwargs)
+                except Exception as e:
+                    error = e
+                    _logger.warning(f"Attempt {i+1} failed with error: {e}")
+                    time.sleep(3)
+            raise Exception(
+                f"Test {test_func} failed after {max_tries} attempts with error: {error!r}"
+            )
+
+        return decorated_func
```

Comment:
```suggestion
    def flaky_test_func(test_func):
        @wraps(test_func)
        def decorated_func(*args, **kwargs):
            error = None
            for i in range(max_tries):
                try:
                    return test_func(*args, **kwargs)
                except Exception as e:
                    if i == max_tries - 1:
                        raise
                     error = e
                    _logger.warning(f"Attempt {i+1} failed with error: {e}")
                    time.sleep(3)

        return decorated_func
```

Can we raise the error here to show the original error?